core/network/network_test.go

```diff
@@ -34,3 +36,24 @@ func TestDedupAddrs(t *testing.T) {
 		})
 	}
 }
+
+func BenchmarkDedupAddrs(b *testing.B) {
+	b.ReportAllocs()
+	var addrs []ma.Multiaddr
+	r := rand.New(rand.NewSource(1234))
+	for i := 0; i < 100; i++ {
+		tcpAddr := ma.StringCast(fmt.Sprintf("/ip4/127.0.0.1/tcp/%d", r.Intn(math.MaxUint16)))
```

Comment:
@sukunrt I tweaked your benchmark function a bit, to make sure the addresses aren't already ordered.